### PR TITLE
Add expired message filter

### PR DESCRIPTION
**Pull request by kadencewp**
Add missing filter for expired key

### CODE REVIEW

**Review by defunctl — CHANGES_REQUESTED** [review]
I think something with sprintf might be a lot easier to read/adjust in the future?

Maybe something like this (untested)?

Edited:

```php
$message_link = apply_filters(
	'stellarwp/uplink/' . Config::get_hook_prefix() . '/messages/expired_key_link',
	'https://evnt.is/195y'
);

$renew_label = apply_filters(
	'stellarwp/uplink/' . Config::get_hook_prefix() . '/messages/renew_label',
	__( 'Renew Your License Now', '%TEXTDOMAIN%' )
);

$opens_in_new_window = apply_filters(
	'stellarwp/uplink/' . Config::get_hook_prefix() . '/messages/screen_reader_text',
	__( ' (opens in a new window)', '%TEXTDOMAIN%' )
);

$notice_text = apply_filters(
	'stellarwp/uplink/' . Config::get_hook_prefix() . '/messages/expired_key_notice',
	__( 'Your license is expired', '%TEXTDOMAIN%' )
);

$message_content = sprintf(
	'%s <a href="%s" target="_blank" class="button button-primary">%s<span class="screen-reader-text">%s</span></a>',
	esc_html( $notice_text ),
	esc_url( $message_link ),
	esc_html( $renew_label ),
	esc_html( $opens_in_new_window )
);

return sprintf(
	'<div class="notice notice-warning"><p>%s</p></div>',
	$message_content
);

```

**Comment by kadencewp** [comment]
@defunctl Well I feel like being able to filter the whole thing is nice because then if I want two buttons or I don't want a link at all I can do that. Only filtering parts feels really limiting. I updated the code let me know what you think of that

File: src/Uplink/Messages/Expired_Key.php [hunk 1]
```diff
@@ -1,20 +1,80 @@
...
+		} else {
+			$message_content = sprintf(
+				'<p>%s</p>',
+				esc_html( $notice_text )
+			);
+		}
+		
+		$message_content = apply_filters( 'stellarwp/uplink/' . Config::get_hook_prefix() . '/messages/expired_key', $message_content );
+		
+		$allowed_html = array(
```
Comment (defunctl) [inline src/Uplink/Messages/Expired_Key.php:55]: Let's use the modern `[]` array syntax for all of these.

File: tests/wpunit/Admin/NoticeTest.php [hunk 2]
```diff
@@ -19,7 +19,7 @@ public function test_it_should_display_notice() {
 		$notice = new Notice();
 		$notice->add_notice( Notice::EXPIRED_KEY, 'uplink' );
 
-		$this->expectOutputString( '<div class="notice notice-warning"><p>Your license is expired<a href="https://evnt.is/195y" target="_blank" class="button button-primary">Renew Your License Now<span class="screen-reader-text"> (opens in a new window)</span></a></p>    </div>' );
+		$this->expectOutputString( '<div class="notice notice-warning"><p>Your license is expired</p></div>' );
```
Comment (defunctl) [inline tests/wpunit/Admin/NoticeTest.php:22]: Shouldn't this still output the original markup unless you've filtered it?

Like, won't this break for everyone else?

**Review by defunctl — CHANGES_REQUESTED** [review]

File: tests/wpunit/Admin/NoticeTest.php (hunk 2 above)
Comment (kadencewp) [reply tests/wpunit/Admin/NoticeTest.php:22]: It's a bug that this outputs with a TEC link as the default. It really shouldn't and shouldn't require a custom filter in every plugin just to fix this. I'm only leaving the link for backwards compatibility with TEC plugins using uplink.

Comment (kadencewp) [reply tests/wpunit/Admin/NoticeTest.php:22]: I'll get another test in tomorrow

Comment (defunctl) [reply tests/wpunit/Admin/NoticeTest.php:22]: Gotcha, I understand now. Yeah, I think one more test and this should be good to go 🚀

**Review by defunctl — APPROVED** [review]
:rocket:

**Pull request merged**